fission-web-api/library/Fission/Web/API/User/DID/Types.hs

```diff
@@ -1,18 +1,30 @@
-module Fission.Web.API.User.DID.Types (DID) where
+module Fission.Web.API.User.DID.Types (DID, SetAuthenticated, SetViaChallenge) where
 
+import           Fission.Challenge.Types
 import qualified Fission.Key                as Key
+import           Fission.User.Username
 
 import           Fission.Web.API.Prelude    hiding (Set)
 
 import qualified Fission.Web.API.Auth.Types as Auth
 
-type DID = "did" :> Set
+type DID = "did" :> (SetAuthenticated :<|> SetViaChallenge)
 
-type Set
+type SetAuthenticated
   =  Summary "Update Public Key"
   :> Description "Set currently authenticated user's root public key to another one"
   --
   :> ReqBody '[JSON] Key.Public
   --
   :> Auth.HigherOrder
   :> PutNoContent
+
+type SetViaChallenge
+  =  Summary "Update Public Key via Email Challenge"
+  :> Description "Set given username's public key to something else via providing a challenge obtained from /user/email/recover"
+  --
+  :> ReqBody '[JSON] Key.Public
+  :> Capture "Username" Username
+  :> QueryParam' '[Required, Strict] "challenge" Challenge
```

Comment:
Why as a query param in particular?

Reply:
Uh, good question. No particular reason honestly. And looking at the other verification route, it looks like it'd be more consistent if I'd be using a path here instead. I'll change that :+1: